File: ui/app/types/ProxyRoute.ts
// Proxy route types matching backend DTOs

export interface PathPrefixConfig {
  enabled: boolean;
  prefix?: string;
  rewriteUrls: boolean;
  rewriteWebSocket: boolean;
  url?: string;
}

export interface SubdomainConfig {
  enabled: boolean;
  subdomain?: string;
  url?: string;
}

export interface PortConfig {
  enabled: boolean;
  port?: number;
  url?: string;
}

export interface IframeConfig {
  enabled: boolean;
  stripXFrameOptions: boolean;
  embedUrl?: string;
}

export interface ProxyUrls {
  pathPrefix?: string;
  subdomain?: string;
  port?: string;
  iframe?: string;
  recommended?: string;
}

export interface ProxyRoute {
  id: number;
  name: string;
  description?: string;
  icon?: string;
  targetUrl: string;
  
  // Access methods
  pathPrefix?: PathPrefixConfig;
  subdomain?: SubdomainConfig;
  port?: PortConfig;
  iframe?: IframeConfig;
  
  // Security
  requireAuth: boolean;
  allowedRoles?: string[];
  
  // Status
  isEnabled: boolean;
  isHealthy: boolean;
  lastHealthCheck?: string;
  
  // Metadata
  createdAt: string;
  updatedAt?: string;
  
  // Generated URLs
  urls?: ProxyUrls;
}

export interface CreateProxyRouteDto {
  name: string;
  description?: string;
  icon?: string;
  targetUrl: string;
  
  // Path Prefix Mode
  enablePathPrefix: boolean;
  pathPrefix?: string;
  rewriteUrls: boolean;
  rewriteWebSocket: boolean;
  
  // Subdomain Mode
  enableSubdomain: boolean;
  subdomain?: string;
  
  // Port Mode
  enablePort: boolean;
  proxyPort?: number;
  
  // Iframe Mode
  enableIframe: boolean;
  stripXFrameOptions: boolean;
  
  // Security
  requireAuth: boolean;
  allowedRoles?: string[];
  
  // Advanced
  timeoutSeconds: number;
  preserveHostHeader: boolean;
  customHeaders?: Record<string, string>;
}

export interface ProxySettings {
  baseDomainType: 'nip.io' | 'sslip.io' | 'custom';
  customBaseDomain?: string;
  portRangeStart: number;
  portRangeEnd: number;
  defaultRequireAuth: boolean;
  serverIp?: string;
  detectedServerIp?: string;
  usedPorts: number[];
}

export interface UpdateProxySettingsDto {
  baseDomainType: 'nip.io' | 'sslip.io' | 'custom';
  customBaseDomain?: string;
  portRangeStart: number;
  portRangeEnd: number;
  defaultRequireAuth: boolean;
  serverIp?: string;
}

export interface ProxyHealthCheck {
  isHealthy: boolean;
  statusCode?: number;
  responseTimeMs: number;
  message?: string;
  error?: string;
}

File: api/Innovatek.Parallel.MiniCluster.Core/Entities/ProxyRoute.cs
using System;
using System.Collections.Generic;

namespace Innovatek.Parallel.MiniCluster.Core.Entities
{
    public class ProxyRoute
    {
        public int Id { get; set; }

        // Basic Info
        public string Name { get; set; } = string.Empty;
        public string? Description { get; set; }
        public string? Icon { get; set; }
        public string TargetUrl { get; set; } = string.Empty;

        // Path Prefix Mode
        public bool EnablePathPrefix { get; set; }
        public string? PathPrefix { get; set; }
        public bool RewriteUrls { get; set; } = true;
        public bool RewriteWebSocket { get; set; }

        // Subdomain Mode
        public bool EnableSubdomain { get; set; } = true;
        public string? Subdomain { get; set; }

        // Port Mode
        public bool EnablePort { get; set; }
        public int? ProxyPort { get; set; }

        // Iframe Mode
        public bool EnableIframe { get; set; }
        public bool StripXFrameOptions { get; set; } = true;

        // Security
        public bool RequireAuth { get; set; } = true;
        public string? AllowedRoles { get; set; }
        public string? ApiKey { get; set; } // Optional API key for authentication

        // Advanced Options
        public bool PreserveHostHeader { get; set; }
        public int TimeoutSeconds { get; set; } = 30;
        public string? CustomHeaders { get; set; } // JSON serialized Dictionary<string, string>

        // Health Status
        public bool IsHealthy { get; set; } = true;
        public DateTime? LastHealthCheck { get; set; }

        // Metadata
        public bool IsEnabled { get; set; } = true;
        public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
        public DateTime? UpdatedAt { get; set; }
    }
}
